tests/PaymentGatewayConstructorTest.php

```diff
@@ -0,0 +1,231 @@
+        /// Test $gatewayUrl value
+        // Test $gatewayUrl parameter with null value
+        array(
+            'apiUrl' => 'http://api.url',
+            'gatewayUrl' => null, //The parameter is being tested
+            'apiKey' => 'apikey',
+            'mobile' => '09100000000',
+            'isAssertion' => false,
+            'test'=>'Test $gatewayUrl parameter with null value',
+        ),
+        // Test $gatewayUrl parameter length more than 512 character
+        array(
+            'apiUrl' => 'http://api.url',
+            'gatewayUrl' => 'https://www.api.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=1&ved=2ahUKEwi56MDUi_zdAhUGiSwKHAwQFjAAegQICRAB&url=http%3A%2F%2Fstring-functions.com%2Flength.aspx&usg=AOvVaw1utca5OnUZuOIkPiOsOGnu&sa=t&rct=j&q=&esrc=s&source=web&cd=1&ved=2ahUKEwi56MDUi_zdAhUGiSwKHc47DAwQFjAAegQICRAB&url=http%3A%2F%2Fstring-functions.com%2Flength.aspx&usg=AOvVaw1utca5OnUZuOIkPiOsOGnu&sa=t&rct=j&q=&esrc=s&source=web&cd=1&ved=2ahUKEwi56MDUi_zdAhUGiSwKHc47DAwQFjAAegQICRAB&url=http%3A%2F%2Fstring-functionscom%2Flength.aspx&usg=AOvVaw1utca5OnUZuOIkPiOsOGnu', //The parameter is being tested
+            'apiKey' => 'apikey',
+            'mobile' => '09100000000',
+            'isAssertion' => false,
+            'test'=>'Test $gatewayUrl parameter length more than 512 character',
+        ),
+        // Test $gatewayUrl parameter with incorrect protocol
+        array(
+            'apiUrl' => 'http://api.url',
+            'gatewayUrl' => 'httpq://gateway.com', //The parameter is being tested
+            'apiKey' => 'apikey',
+            'mobile' => '09100000000',
+            'isAssertion' => false,
+            'test'=>'Test $gatewayUrl parameter with incorrect protocol'
+        ),
+        // Test $gatewayUrl parameter  without protocol
+        array(
+            'apiUrl' => 'http://api.url',
+            'gatewayUrl' => 'gateway.url', //The parameter is being tested
+            'apiKey' => 'apikey',
+            'mobile' => '09100000000',
+            'isAssertion' => false,
+            'test'=>'Test $gatewayUrl parameter without protocol'
+        ),
+        // Test $gatewayUrl parameter with port
+        array(
+            'apiUrl' => 'https://api.url',
+            'gatewayUrl' => 'https://api.com:8080', //The parameter is being tested
+            'apiKey' => 'apikey',
+            'mobile' => '09100000000',
+            'isAssertion' => false,
+            'test'=>'Test $gatewayUrl parameter with port'
+        ),
+        // Test $gatewayUrl parameter with Ip
+        array(
+            'apiUrl' => 'https://api.url',
+            'gatewayUrl' => 'https://80.80.80.80', //The parameter is being tested
+            'apiKey' => 'apikey',
+            'mobile' => '09100000000',
+            'isAssertion' => true,
+            'test'=>'Test $gatewayUrl parameter with Ip'
+        ),
+        // Test $gatewayUrl parameter with Ip and port
+        array(
+            'apiUrl' => 'https://api.url',
```

Comment:
api url have to convert to variable that load from property file.

Reply:
this is just sample url  for test constructor validation url